include/linux/soundwire/sdw_intel.h

```diff
@@ -119,8 +119,7 @@ struct sdw_intel_ops {
 			     struct sdw_intel_stream_params_data *params_data);
 	int (*free_stream)(struct device *dev,
 			   struct sdw_intel_stream_free_data *free_data);
-	int (*trigger)(struct snd_pcm_substream *substream,
-		       int cmd, struct snd_soc_dai *dai);
+	int (*trigger)(int stream, int cmd, struct snd_soc_dai *dai);
```

Comment:
If we do this, then probably the `dai` should be the first parameter?

Reply:
> If we do this, then probably the `dai` should be the first parameter?

ack

Reply:
BTW, do you have better naming for `ipc4_sdw_trigger`?

Reply:
ipc4_sdw_trigger is confusing, since the actual trigger is done at the dailink level using the bank switch. This trigger is really about pipeline management, not actual SoundWire trigger.

maybe ipc4_sdw_dai_update_state_on_trigger ?